### PR TITLE
Taskcluster: add dev build and unit tests on macOS

### PR DESCRIPTION
**Pull request by SimonSapin**
For simplicity, Salt is used in [agentless](https://docs.saltstack.com/en/getstarted/ssh/) mode for now.

I’m also hoping to have most of the dependencies be installed by tasks for themselves, so that changes can be made in the same `servo/servo` PR that uses them without redeploying servers.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22185)

### CODE REVIEW

File: etc/taskcluster/macos/states/generic-worker.sls [hunk 1]
```diff
@@ -16,21 +17,76 @@
...
-    - creates: {{ keyfile }}
-    - prepend_path: {{ bin }}
+    - creates: {{ home }}/key
+    - runas: worker
+
+/Library/LaunchAgents/net.generic.worker.plist:
+  file.managed:
+    - mode: 644
+    - template: jinja
+    - contents: >-
```
Comment (Manishearth) [inline etc/taskcluster/macos/states/generic-worker.sls]: is it possible to store this file separately?

<s>i suppose the substs make this harder</s>

nah, we use files with substs in our current salt setup: https://github.com/servo/saltfs/blob/master/homu/init.sls#L69

File: etc/taskcluster/macos/states/generic-worker.sls [hunk 2]
```diff
@@ -44,12 +44,27 @@
...
 
+{{ home }}/run:
+  file.managed:
+    - mode: 744
+    - user: {{ user }}
+    - template: jinja
+    - contents: |-
+        #!/bin/sh
+        # generic-worker overwrites its config file to fill in defaults,
+        # but we want to avoid touching config.json here
```
Comment (Manishearth) [inline etc/taskcluster/macos/states/generic-worker.sls]: try to split this into its own file if possible

**Review by Manishearth — APPROVED** [review]

File: etc/taskcluster/macos/Saltfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+salt-ssh:
+  config_dir: ./config
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply etc/taskcluster/macos/Saltfile]: Done.

File: etc/taskcluster/macos/states/generic-worker.sls (hunk 1 above)
Comment (SimonSapin) [reply etc/taskcluster/macos/states/generic-worker.sls]: Done.

File: etc/taskcluster/macos/states/generic-worker.sls (hunk 2 above)
Comment (SimonSapin) [reply etc/taskcluster/macos/states/generic-worker.sls]: Removed as part of another change.

File: etc/taskcluster/macos/README.md [hunk 4]
```diff
@@ -15,10 +15,20 @@ cd etc/taskcluster/macos
...
+* Change the administrator password to one generated with
+  `</dev/urandom tr -d -c 'a-zA-Z' | head -c 8; echo`
+  (this short because of VNC),
+  and save it in the shared 1Password account.
+
+* Give the public IPv4 address a DNS name through Cloudflare.
+
+* Add a correponding entry in the `config/roster` file.
+
+* Log in through VNC, and run `xcode-select --install`
```
Comment (jdm) [inline etc/taskcluster/macos/README.md]: https://github.com/servo/servo/wiki/SaltStack-Administration has a number of other steps we follow right now.

Comment (SimonSapin) [reply etc/taskcluster/macos/README.md]: I’m hoping to make some of those taken care of by Salt or by tasks, but installing XCode requires GUI interaction.

Comment (SimonSapin) [reply etc/taskcluster/macos/README.md]: Also, this document is intended to expand over time as we do more, it is not at an end state. This is documenting what I’ve done so far and how to reproduce it.

**Review by Manishearth — APPROVED** [review]
r=me

**Comment by SimonSapin** [comment]
```
  ▶ Unexpected subtest result in /html/browsers/browsing-the-web/unloading-documents/unload/004.html:
  │ FAIL [expected PASS] pagehide / unload event order
  │   → assert_array_equals: lengths differ, expected 2 got 3
  │ 
  │ onload/</iframe.onload/iframe.onload<@http://web-platform.test:8000/html/browsers/browsing-the-web/unloading-documents/unload/004.html:19:51
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1568:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1592:20
```

@bors-servo retry

**Pull request merged**